core/src/components/input/input.tsx

```diff
@@ -117,13 +128,45 @@ export class Input implements ComponentInterface {
    */
   @Prop() enterkeyhint?: 'enter' | 'done' | 'go' | 'next' | 'previous' | 'search' | 'send';
 
+  /**
+   * Text that is placed under the input and displayed when an error is detected.
+   */
+  @Prop() errorText?: string;
+
   /**
    * A hint to the browser for which keyboard to display.
    * Possible values: `"none"`, `"text"`, `"tel"`, `"url"`,
    * `"email"`, `"numeric"`, `"decimal"`, and `"search"`.
    */
   @Prop() inputmode?: 'none' | 'text' | 'tel' | 'url' | 'email' | 'numeric' | 'decimal' | 'search';
 
+  /**
+   * Text that is placed under the input and displayed when no error is detected.
+   */
+  @Prop() helperText?: string;
+
+  /**
+   * How to pack the label and the input within a line. This property only applies when the input and label are on the same line. In other words, this property is ignored when `labelPlacement` is set to `'floating'` or `'stacked'`.
+   * `'start'`: The label and input are packed on the left in LTR and on the right in RTL.
+   * `'end'`: The label and input are packed on the right in LTR and on the left in RTL.
+   * `'space-between'`: The label and input are placed at either end of the line with empty space between the elements. Which end each element can be configured using the `'start'` or `'end'` values on the `labelPlacement` property.
+   */
+  @Prop() justify: 'start' | 'end' | 'space-between' = 'start';
+
+  /**
+   * The visible label associated with the input.
+   */
+  @Prop() label?: string;
+
+  /**
+   * Where to place the label relative to the input.
+   * `'start'`: The label will appear to the left of the input in LTR and to the right in RTL.
+   * `'end'`: The label will appear to the right of the input in LTR and to the left in RTL.
+   * `'floating'`: The label will appear smaller and above the input when the input is focused or it has a value. Otherwise it will appear on top of the input.
+   * `'stacked'`: The label will appear smaller and above the input regardless even when the input is blurred or has no value.
+   */
+  @Prop() labelPlacement: 'start' | 'end' | 'floating' | 'stacked' = 'start';
```

Comment:
Open to making this `labelPosition` to keep the API similar to the `ion-label` `position` property

Reply:
I sort of like `labelPlacement` to avoid associations with the CSS `position` property, but I could go either way 😮